src/prefect/cli/cloud/__init__.py

```diff
@@ -424,6 +425,8 @@ async def login(
             )
         except httpx.HTTPStatusError as exc:
             exit_with_error(f"Error connecting to Prefect Cloud: {exc!r}")
+        except (CancelledError, TimeoutError):
```

Comment:
Hmm I'm not quite following why we now need to catch `CancelledError` and `TimeoutError` now.

Shouldn't the end result of retrying 500s unsuccessfully end up raising the 500? (Question for here and below)

Reply:
The problem is this `fail_after`: https://github.com/PrefectHQ/prefect/blob/227dfcc7e3374c212a4bcd68b14e090b1c02d9d3/src/prefect/client/cloud.py

What happens is that because we now retry with exponential backoff, we can fail this timer before reaching our max retries, in which case we see a timeout error (or canceled, I hit both IIRC).